java-checks/src/main/java/org/sonar/java/checks/ArrayDesignatorOnVariableCheck.java

```diff
@@ -42,18 +42,29 @@
 @ActivatedByDefault
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("5min")
-public class ArrayDesignatorOnVariableCheck extends SquidCheck<LexerlessGrammar> implements JavaCheck {
+public class ArrayDesignatorOnVariableCheck extends SubscriptionBaseVisitor {
 
   @Override
-  public void init() {
-    subscribeTo(Kind.VARIABLE);
-    subscribeTo(JavaLexer.VARIABLE_DECLARATOR);
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.VARIABLE);
   }
 
   @Override
-  public void visitNode(AstNode node) {
-    if (node.hasDirectChildren(TreeFactory.WRAPPER_AST_NODE) || (node.is(Kind.VARIABLE) && ((VariableTreeImpl) node).dims() > 0)) {
-      getContext().createLineViolation(this, "Move the array designator from the variable to the type.", node);
+  public void visitNode(Tree tree) {
+    VariableTree variableTree = (VariableTree) tree;
+    TypeTree type = variableTree.type();
+    int identifierColumn = variableTree.simpleName().identifierToken().column();
+    while (type.is(Tree.Kind.ARRAY_TYPE)) {
+      ArrayTypeTree arrayTypeTree = (ArrayTypeTree) type;
+      SyntaxToken arrayDesignatorToken = arrayTypeTree.ellipsisToken();
+      if (arrayDesignatorToken == null) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Redundant nullcheck of arrayDesignatorToken, which is known to be non-null in org.sonar.java.checks.ArrayDesignatorOnVariableCheck.visitNode(Tree) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)